File: backend/app/models/class_model.py
from typing import Optional, List, Dict, Any
from pydantic import BaseModel
from datetime import datetime


class Schedule(BaseModel):
    days: List[str] = []
    time: Optional[str] = None


class ClassBase(BaseModel):
    name: str
    subject: str
    grade: str
    schedule: Optional[Schedule] = None


class ClassCreate(ClassBase):
    pass


class ClassUpdate(BaseModel):
    name: Optional[str] = None
    subject: Optional[str] = None
    grade: Optional[str] = None
    schedule: Optional[Schedule] = None
    is_active: Optional[bool] = None


class ClassResponse(ClassBase):
    id: str
    teacher_id: Optional[str] = None
    student_ids: List[str] = []
    is_active: bool = True
    created_at: Optional[str] = None

    class Config:
        from_attributes = True

File: flutter_app/lib/models/class_model.dart
class ClassModel {
  final String id;
  final String name;
  final String subject;
  final String grade;
  final String? teacherId;
  final List<String> studentIds;
  final Map<String, dynamic> schedule;
  final bool isActive;
  final String? createdAt;

  ClassModel({
    required this.id,
    required this.name,
    required this.subject,
    required this.grade,
    this.teacherId,
    this.studentIds = const [],
    this.schedule = const {},
    this.isActive = true,
    this.createdAt,
  });

  factory ClassModel.fromJson(Map<String, dynamic> json) => ClassModel(
        id: json['id'] ?? '',
        name: json['name'] ?? '',
        subject: json['subject'] ?? '',
        grade: json['grade'] ?? '',
        teacherId: json['teacher_id'],
        studentIds: List<String>.from(json['student_ids'] ?? []),
        schedule: Map<String, dynamic>.from(json['schedule'] ?? {}),
        isActive: json['is_active'] ?? true,
        createdAt: json['created_at'],
      );

  Map<String, dynamic> toJson() => {
        'id': id,
        'name': name,
        'subject': subject,
        'grade': grade,
        'teacher_id': teacherId,
        'student_ids': studentIds,
        'schedule': schedule,
        'is_active': isActive,
        'created_at': createdAt,
      };

  int get studentCount => studentIds.length;
}
